directory/datadog_checks/directory/directory.py

```diff
@@ -54,9 +53,9 @@ def check(self, _):
             # report missing directory
             self.service_check(name=SERVICE_DIRECTORY_EXISTS, status=self.WARNING, tags=service_check_tags, message=msg)
 
-            # raise exception if `ignore_missing` is False
+            # raise warning if `ignore_missing` is False
             if not self._config.ignore_missing:
-                raise CheckException(msg)
+                raise Warning(msg)
```

Comment:
We don't use this anywhere in the agent integrations and I'm not sure what's the goal here. You're changing the agent status from error to warning? why?

Reply:
I think I could raise a `ValueError` or `ConfigurationError` instead of a `Warning` but the thought behind this was that since the [service check](https://github.com/DataDog/integrations-core/blob/7.32.x/directory/datadog_checks/directory/directory.py#L55) and [log message reported](https://github.com/DataDog/integrations-core/blob/7.32.x/directory/datadog_checks/directory/directory.py#L61) in the Agent are both at WARN levels, it seemed inconsistent to have the check status report an Error. I wanted to make the behavior consistent across the integration. The customer in question was aware of the option to `ignore_missing` option and was testing the integration.

![image](https://user-images.githubusercontent.com/37895372/198299806-224a0693-3c4f-46e6-adcc-18cd397c7777.png)

Reply:
ConfigurationError is still going to make it status error but it is not a ConfigurationError, that exception is for things like you have missed a mandatory field in the configuration or you have specified two options that are not compatible with each other etc, this is a runtime error.

To be honest this feature is very weirdly defined and I do not have context on why is it the way it is but for me if `ignore_missing` is true then there should not even be a warning service check at all and then in either case it should not throw an exception at any point but all those things are breaking changes so I'm wary of making that change.